### PR TITLE
Single result queries

### PR DESCRIPTION
**Pull request by Blackbaud-ErikBudtke**
@sophiaso @Incarnation-p-lee please review.

### CODE REVIEW

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/RoleRepositoryCollectionIT.java [hunk 1]
```diff
@@ -37,6 +45,11 @@
     @Autowired
     private DocumentDbTemplate template;
 
+    @Before
+    public void setup() {
```
[the remark this answers is not in the document]
Comment (Blackbaud-ErikBudtke) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/RoleRepositoryCollectionIT.java]: Not sure if this is a valid comment or not, there are instances of both `setup` & `setUp` in the tests.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/SingleResponseRepositoryIT.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestRepositoryConfig.class)
+public class SingleResponseRepositoryIT {
+
+    private static final Contact TEST_CONTACT = new Contact("testId", "faketitle");
+
+    @Autowired
```
[the remark this answers is not in the document]
Comment (Blackbaud-ErikBudtke) [reply src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/integration/SingleResponseRepositoryIT.java]: Also not sure what to make of this, seems as if the bot is complaining on the scoping of the `@Autowired` annotation. Seems like a misconfiguration of the bot @sophiaso @Incarnation-p-lee

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/DocumentDbQueryExecution.java [hunk 3]
```diff
@@ -41,6 +46,38 @@ public Object execute(DocumentQuery query, Class<?> type, String collection) {
...
+
+        public SingleEntityExecution(DocumentDbOperations operations, ReturnedType returnedType) {
+            this.operations = operations;
+            this.returnedType = returnedType;
+        }
+
+        @Override
+        public Object execute(DocumentQuery query, Class<?> type, String collection) {
+            final List results = operations.find(query, type, collection);
+            Object result;
```
[the remark this answers is not in the document]
Comment (Blackbaud-ErikBudtke) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/DocumentDbQueryExecution.java]: This can't be final...

**Comment by Blackbaud-ErikBudtke** [comment]
@Incarnation-p-lee the Codacy step is blocking this PR, but as stated in my comments, I'm not 100% sure they are valid. Let me know if there is something we can take a look at.

**Comment by Incarnation-p-lee** [comment]
Could you please fix some CI reported issue ? So we can start the review process. For `setUp` or `setup` issue, please take the advice from `codacy`.

**Comment by Blackbaud-ErikBudtke** [comment]
Codacy issues fixed, looks like the CI is failing with some authorization issues. Let me know if there is something that needs to change on our end.

**Comment by kushagraThapar** [comment]
@Blackbaud-ErikBudtke can you please resolve the conflicts ?
I am looking at this PR.

**Comment by Blackbaud-ErikBudtke** [comment]
Closing this in favor of https://github.com/microsoft/spring-data-cosmosdb/pull/480

**Pull request closed without merging**